app/scripts/stores/dashboardStore.js

```diff
@@ -0,0 +1,290 @@
+                    this.downloadedItems.delete(parentId)
+                    this.downloadedItems.set(parentId, downloadedParent)
+                }
+                mainStore.listItems = parentsChildren
+                this.responseHeaders = headers;
+                mainStore.loading = false;
+            }).catch(ex =>this.handleErrors(ex))
+    }
+
+    @action toggleDrawer() {
+        let drawerPosition = !this.drawer.get('open')
+        let contentStyle = { transition: 'margin-left 450ms cubic-bezier(0.23, 1, 0.32, 1)' };
+        drawerPosition ? contentStyle.marginLeft = this.drawer.get('width') : null
+        this.drawer.set('open', drawerPosition)
+        this.drawer.set('contentStyle', contentStyle)
+    }
+    
+    @action setDrawer() {
+        let width = 350;
+        let contentStyle = { transition: 'margin-left 450ms cubic-bezier(0.23, 1, 0.32, 1)' };
+        contentStyle.marginLeft = width;
+        this.drawer.set('open', true);
+        this.drawer.set('width', width);
+        this.drawer.set('contentStyle', contentStyle);
+    }
+    
+    @action toggleCollapseTree(router) {
+        if (this.drawer.get('collapsed')) {
+            this.downloadedItems.forEach((item) => {
+                item.open = true
+            })
+            this.drawer.set('collapsed', false);
+            this.drawer.set('toggleLable', 'Home');
+        } else {
+            mainStore.listItems = mainStore.projects
+            this.selectedItem = null
+            this.downloadedItems.forEach((item) => {
+                item.open = false
+            })
+            router.push({pathname: ("/dashboard")})
+            this.drawer.set('collapsed', true);
+            this.drawer.set('toggleLable', 'Expand');
+        }
+    }
+    
+    @action toggleTreeListItem(id) {
+        this.selectedItem = id;
+        let item = this.downloadedItems.get(id)
+        item.open = !item.open
+        this.downloadedItems.delete(id)
+        this.downloadedItems.set(id, item)
+    }
+
+    @action getItem(id, path, isSelected) {
+        mainStore.loading = true;
+        this.transportLayer.getEntity(id, path)
+            .then(checkStatusAndConsistency)
+            .then(response => response.json())
+            .then((json) => {
+                if(!json.error) {
+                    let item = this.downloadedItems.get(id)
+                    if(item) {
+                        json.open = true
+                        json.childrenIds = item.childrenIds
+                        json.folderIds = item.folderIds
+                        this.downloadedItems.delete(id)
+                    }
+                    this.downloadedItems.set(id, json)
+                    if(isSelected) this.selectedItem = id
+                    mainStore.loading = false;
+                } else {
+                    json.code === 'resource_not_consistent' ? mainStore.tryAsyncAgain(getItem, [id, path, isSelected]) : mainStore.handleErrors(json);
```

Comment:
I think that you're missing some params for `tryAsyncAgain()`

Should have `tryAsyncAgain(func, args, delay, counterId, message, isUpload)`

1.) function to retry
2.) arguments array that the function to be retried needs
3.) A delay in milliseconds
4.)An ID for the counter (that way each counter is unique) you can just use the `id`of the item
5.) A message to display to the user
6.) whether or not it's an upload (bool)

Reply:
These are great notes, Thanks
I will work on this on Monday

Reply:
I approved this PR but just realized that these changes still need to be made.